my-app/src/app/modules/courses-page/courses-page.component.spec.ts

```diff
@@ -1,40 +1,73 @@
 import {ComponentFixture, TestBed} from '@angular/core/testing';
-import {NO_ERRORS_SCHEMA} from "@angular/core";
+import {Component, EventEmitter, Input, NO_ERRORS_SCHEMA, Output} from "@angular/core";
 
 import {COURSE_DATA} from "@assets/mocks/course-data.mock";
 
-import {FilterPipe, OrderByPipe} from "./core";
+import {By} from "@angular/platform-browser";
+
+import {FilterPipe, OrderByPipe} from "./pipes";
 
 import {CoursesPageComponent} from './courses-page.component';
+import {CoursesHandlerService} from "./services/courses-handler.service";
+
+import {CoursesComponent} from "./components";
 
 import Spy = jasmine.Spy;
 
+@Component({
+  selector: 'app-courses',
+  template: '',
+})
+class FakeCoursesComponent implements Partial<CoursesComponent> {
+  @Input()
+  public course = COURSE_DATA[0];
+
+  @Output()
+  public deleteCourse = new EventEmitter<number>();
+}
+
 describe('CoursePageComponent', () => {
   let app: CoursesPageComponent;
   let fixture: ComponentFixture<CoursesPageComponent>;
+  let coursesHandlerService: CoursesHandlerService;
+  let coursesComponent: FakeCoursesComponent;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [CoursesPageComponent, OrderByPipe, FilterPipe],
+      declarations: [
+        CoursesPageComponent,
+        OrderByPipe,
+        FilterPipe,
+        FakeCoursesComponent
+      ],
       schemas: [NO_ERRORS_SCHEMA]
     }).compileComponents();
 
+    coursesHandlerService = TestBed.inject(CoursesHandlerService);
```

Comment:
Usually it's a bad idea to use real instance of a service. Prefer mocking e.g. using `providers` option with useClass.

Reply:
Added mock files for all services